js-release/src/index.ts

```diff
@@ -0,0 +1,127 @@
+import { exec } from "child_process";
+import readdirp from "readdirp";
+import * as core from "@actions/core";
+import { Storage } from "@google-cloud/storage";
+import setupGCloud from "./setup-gcloud/setup-gcloud";
+
+// config
+const BUCKET = "manifold-js";
+const URL_MAP = "manifold-js-cdn-urlmap";
+
+/**
+ * Publish to npm
+ */
+async function npmPublish({
+  npmDistTag,
+  npmToken,
+  workspace,
+}: {
+  npmDistTag: string;
+  npmToken: string;
+  workspace: string;
+}) {
+  await exec(`echo "//registry.npmjs.org/:_authToken=${npmToken}" > ~/.npmrc`);
+  await exec(`npm publish ${workspace} --tag ${npmDistTag}`);
+}
+
+/**
+ * Publish to Manifold CDN
+ */
+async function cdnPublish({
+  packageName,
+  semver,
+  serviceAccountKey,
+  workspace,
+}: {
+  packageName: string;
+  semver: string;
+  serviceAccountKey: string;
+  workspace: string;
+}) {
+  // recursively read workspace dir
+  const files = await readdirp.promise(workspace);
+
+  // upload files to CDN
+  const storage = new Storage({ keyFile: serviceAccountKey });
+  files.forEach((file) => {
+    storage.bucket(BUCKET).upload(file.fullPath, {
+      destination: `${packageName}@${semver}/${file.path}`,
+    });
+  });
+
+  // update URL Map
+  await setupGCloud();
+  // latest
+  await exec(
+    `gcloud compute url-maps remove-path-matcher ${URL_MAP} --path-matcher-name=latest`
```

Comment:
```suggestion
    `gcloud compute url-maps remove-path-matcher ${URL_MAP} --path-matcher-name=${packageName}-latest`
```
And so on for the rest of the ones below otherwise the pathc matcher names are going to conflict when we have multiple packages using this action